components/builder-api/src/http/handlers.rs

```diff
@@ -145,24 +147,39 @@ pub fn job_group_promote(req: &mut Request) -> IronResult<Response> {
         Err(err) => return Ok(render_net_error(&err)),
     };
 
-    // JB TODO - This logic was mostly ported here from the scheduler, so that the scheduler can
-    // now just make 1 HTTP call to promote a group, instead of N calls for every project in the
-    // group. But, it does seem odd for the scheduler to be making HTTP calls to builder-api when
-    // we have a zmq API that's much faster for use between services.
-    for project in group.get_projects().into_iter().filter(|x| {
-        x.get_state() == ProjectState::Success
-    })
-    {
-        let ident = OriginPackageIdent::from_str(project.get_ident()).unwrap();
+    // This only makes sense if the group is complete. If the group isn't complete, return now and
+    // let the user know.
+    if group.get_state() != GroupState::Complete {
+        return Ok(Response::with((status::Forbidden, "rg:jgp:1")));
```

Comment:
I have no idea what `rg` stands for.  I cargo-culted it from elsewhere in this file, but I figured it was good to be consistent 😁